nsx/resource_nsx_edge_dhcp.go

```diff
@@ -0,0 +1,358 @@
+package nsx
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"log"
+	"net"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/IBM-tfproviders/govnsx"
+	"github.com/IBM-tfproviders/govnsx/nsxresource"
+	"github.com/IBM-tfproviders/govnsx/nsxtypes"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+type ipRange struct {
+	startIP string
+	endIP   string
+}
+
+type dhcpCfg struct {
+	edgeId      string
+	portgroup   string
+	networkAddr string // network address eg: 10.10.10.0
+	netMask     string // netmask eg: 255.255.255.0
+	vnicAddr    string
+	defaultGw   string
+	ipRange
+}
+
+func resourceNsxEdgeDHCP() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxEdgeDHCPCreate,
+		Read:   resourceNsxEdgeDHCPRead,
+		Update: resourceNsxEdgeDHCPUpdate,
+		Delete: resourceNsxEdgeDHCPDelete,
+
+		Schema: map[string]*schema.Schema{
+			"cidr": &schema.Schema{
+				Type:         schema.TypeString,
```

Comment:
This looks to hold only one subnet. Can we have  a array/map call "subnets" and have cidr, gw, ipool range under it.
Also, any plan to support other dhcp attributes such as dns etc,..